src/main/java/uk/sky/cqlmigrate/LockConfig.java

```diff
@@ -68,8 +75,18 @@ public LockConfigBuilder withTimeout(Duration timeout) {
             return this;
         }
 
+        /**
+         * Release the lock in case of failure
```

Comment:
This implies that the default behaviour is that it wont release the lock on failure. But maybe it's worth adding it here explicitly and saying why that's default?